projectview/src/test/java/org/jetbrains/bsp/bazel/projectview/parser/sections/ProjectViewSingletonSectionParserTest.java

```diff
@@ -0,0 +1,221 @@
+        sectionTry.getCause().getMessage());
+  }
+
+  @Test
+  public void shouldReturnEmptyForEmptySectionBody() {
+    // given
+    var rawSection = new ProjectViewRawSection(sectionName, "");
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    assertFalse(section.isPresent());
+  }
+
+  @Test
+  public void shouldReturnSectionWithTrimmedValue() {
+    // given
+    var rawSection = new ProjectViewRawSection(sectionName, "  value");
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    var expectedSection = sectionConstructor.apply("value");
+    assertEquals(expectedSection, section.get());
+  }
+
+  @Test
+  public void shouldReturnSectionWithTrimmedValueWithSpaces() {
+    // given
+    var rawSection = new ProjectViewRawSection(sectionName, "  value with space 123 \t\n");
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    var expectedSection = sectionConstructor.apply("value with space 123");
+    assertEquals(expectedSection, section.get());
+  }
+
+  // T parse(rawSections)
+
+  @Test
+  public void shouldReturnLastSectionWithoutExplicitDefault() {
+    // given
+    var rawSection1 = new ProjectViewRawSection("anotersection1", "value1");
+    var rawSection2 = new ProjectViewRawSection(sectionName, "  value2\n");
+    var rawSection3 = new ProjectViewRawSection("anotersection2", "\tvalue3\n");
+    var rawSection4 = new ProjectViewRawSection(sectionName, "    value4\n  ");
+    var rawSection5 = new ProjectViewRawSection("anotersection3", "\tvalue5\n");
+
+    var rawSections =
+        new ProjectViewRawSections(
+            List.of(rawSection1, rawSection2, rawSection3, rawSection4, rawSection5));
+
+    // when
+    var section = parser.parse(rawSections);
+
+    // then
+    var expectedSection = sectionConstructor.apply("value4");
+    assertEquals(expectedSection, section.get());
+  }
+
+  @Test
+  public void shouldReturnEmptyIfSectionDoesntExist() {
+    // given
+    var rawSection1 = new ProjectViewRawSection("anotersection1", "value1");
+    var rawSection2 = new ProjectViewRawSection("anotersection2", "  value2");
+    var rawSection3 = new ProjectViewRawSection("anotersection3", "\tvalue3\n");
+
+    var rawSections = new ProjectViewRawSections(List.of(rawSection1, rawSection2, rawSection3));
+
+    // when
+    var section = parser.parse(rawSections);
+
+    // then
+    assertFalse(section.isPresent());
+  }
+
+  // T parseOrDefault(rawSections, defaultValue)
+
+  @Test
+  public void shouldReturnLastSection() {
+    // given
+    var rawSection1 = new ProjectViewRawSection("anotersection1", "value1");
+    var rawSection2 = new ProjectViewRawSection(sectionName, "  value2\n");
```

Comment:
Looks like the idea is that `sectionName` holds a valid section name, whereas `anotersection*` represent invalid ones. If that's correct, maybe rename these to make that more obvious.

Reply:
`anotersection*` represents another section, it can be valid (it will be parsed by some parser) but it is not the one we are looking for